packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.ts

```diff
@@ -75,4 +97,88 @@ export class ApplicationExtension<
         const {allRoutes} = params
         return allRoutes
     }
+
+    /**
+     * Called on the server to serialize the extension data that will be sent to the client.
+     *
+     * @returns SerializedExtension - The serialized extension data.
+     * @throws Error if the routes have not been loaded.
+     */
+    public serialize(): SerializedExtension {
+        if (this._cachedRoutes === null) {
+            throw new Error('Routes have not been loaded. Call getRoutes() before serializing')
+        }
+        const serializedRoutes = this._cachedRoutes.map((route) => {
+            // Check if it is already serialized
+            if ('componentName' in route) {
+                return route
+            }
```

Comment:
It looks like with @vmarta 's changes after calling `getRoutesAsync` some routes may already be a `SerializedRoute`. This is because they have a `componentName` instead of a `component` prior to `beforeRouteMatch`. In this case we just return the `SerializedRoute` as is